src/sync/browser.js

```diff
@@ -0,0 +1,105 @@
+import PushManagerFactory from './PushManager';
+import { PUSH_DISCONNECT, PUSH_CONNECT } from './constants';
+import FullProducerFactory from '../producer';
+import PartialProducerFactory from '../producer/browser/Partial';
+import { matching } from '../utils/key/factory';
+import { forOwn } from '../utils/lang';
+import logFactory from '../utils/logger';
+const log = logFactory('splitio-sync:sync-manager');
+
+/**
+ * Factory of sync manager for browser
+ *
+ * @param context main client context
+ */
+export default function BrowserSyncManagerFactory(mainContext) {
+
+  const contexts = {};
+  let pushManager;
+
+  function startPolling() {
+    log.info('PUSH down or disconnected. Starting periodic fetch of data.');
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      if (producer && !producer.isRunning())
+        producer.start(true); // `fetchers` are scheduled but not called immediately
+    });
+  }
+
+  function stopPollingAndSyncAll() {
+    log.info('PUSH (re)connected. Syncing and stopping periodic fetch of data.');
+    // if polling, stop
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      if (producer && producer.isRunning())
+        producer.stop();
+    });
+    syncAll();
+  }
+
+  function syncAll() {
+    // fetch splits and segments
+    const mainProducer = mainContext.get(mainContext.constants.PRODUCER, true);
+    mainProducer && mainProducer.synchronizeSplits();
+    forOwn(contexts, function (context) {
+      const producer = context.get(context.constants.PRODUCER, true);
+      producer && producer.synchronizeMySegments();
+    });
+  }
+
+  function shared(context, isSharedClient = true) {
+    const producer = isSharedClient ? PartialProducerFactory(context) : FullProducerFactory(context);
+    context.put(context.constants.PRODUCER, producer);
+    const settings = context.get(context.constants.SETTINGS);
+    const userKey = matching(settings.core.key);
+    if(contexts[userKey]) log.warn('A client with the same user key has already been created. Only the new instance will be properly synchronized.');
```

Comment:
keep your spacing consistent. Both in line jumps and space before the parenthesis. Let's schedule a task later to review the linting and adjust some light autofix rules at some point of Q1.